app/services/person_constructor.rb

```diff
@@ -74,9 +74,7 @@ def strip_whitespace_from_values
   end
 
   def normalize_zip_code
-    if params[:zip_code]
-      params[:zip_code] = params[:zip_code].gsub!(/\A(\d{5})[^\w]?(\d{4})?\z/, '\1')
-    end
+    params[:zip_code].try(:gsub!, /\A(\d{5})[^\w]?(\d{4})\z/, '\1')
```

Comment:
this code fixes the issues that I had before. I'm still open to talking about moving this to a `before_validation` callback on `Location`, though. Also would be open to going back to using the `ZipCode.valid_zip_5` method that was removed a while back.